tests/unit/test_charm.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 dima.tisnek@canonical.com
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+import pytest
+from ops.testing import Harness
+from ops.model import ActiveStatus, MaintenanceStatus
+from charm import HexanatorCharm
+
+
+@pytest.fixture
+def harness():
+    harness = Harness(HexanatorCharm)
+    harness.begin()
+    yield harness
+    harness.cleanup()
+
+
+def test_pebble_ready(harness):
+    assert harness.model.unit.status == MaintenanceStatus('')
+
+    harness.container_pebble_ready('gubernator')
+
+    assert harness.model.unit.status == ActiveStatus()
+
+
+def test_ingress_requirer(harness):
+    assert harness.charm.ingress._auto_data == (None, None, 80)
```

Comment:
Is there another way to test this without resorting to accessing lib-private fields? Could we listen for the custom events the charm lib sends instead? (If any.)

Reply:
I think the proper way is to examine the data bag.
`test_scenario.py: test_ingress()` does that.

I'm not sure about harness tests, I feel they are meant to be lower level, but maybe I'm wrong and should send an relation event and then poke at the databag too.

Reply:
It is definitely common for Harness tests to trigger relation events (either explicitly or via the shortcuts) and to assert on what's in the data bags.